File: repos_estudiantes_ALVAREZ_segunda_entrega/Calle_Castillo_Cristhian_Javier/Practica 2 prog-121/AGREGACION Y COMPOSICION/EJERCICIO 3/Avion.py
class Parte:
    def __init__(self, nombre, peso):
        self._nombre = nombre
        self._peso = peso

    # Getters
    def get_nombre(self):
        return self._nombre

    def get_peso(self):
        return self._peso

    # Setters
    def set_nombre(self, nombre):
        self._nombre = nombre

    def set_peso(self, peso):
        self._peso = peso

    # Método para mostrar información
    def mostrar_info(self):
        print(f"Parte: {self._nombre}, Peso: {self._peso} kg")


class Avion:
    def __init__(self, modelo, fabricante):
        self._modelo = modelo
        self._fabricante = fabricante
        self._partes = []

    # Getters
    def get_modelo(self):
        return self._modelo

    def get_fabricante(self):
        return self._fabricante

    def get_partes(self):
        return self._partes

    # Setters
    def set_modelo(self, modelo):
        self._modelo = modelo

    def set_fabricante(self, fabricante):
        self._fabricante = fabricante

    # Método para agregar una parte
    def agregar_parte(self, parte):
        if isinstance(parte, Parte):
            self._partes.append(parte)

    # Método para mostrar información del avión
    def mostrar_avion(self):
        print(f"Modelo del avión: {self._modelo}")
        print(f"Fabricante: {self._fabricante}")
        print("Partes del avión:")
        for p in self._partes:
            p.mostrar_info()

if __name__ == "__main__":
    # Crear partes del avión
    motor = Parte("Motor", 1200)
    ala_izquierda = Parte("Ala Izquierda", 800)
    ala_derecha = Parte("Ala Derecha", 800)
    tren_aterrizaje = Parte("Tren de Aterrizaje", 600)

    avion = Avion("Boeing 737", "Boeing")
    # Agregar partes al avión
    avion.agregar_parte(motor)
    avion.agregar_parte(ala_izquierda)
    avion.agregar_parte(ala_derecha)
    avion.agregar_parte(tren_aterrizaje)

    # Mostrar información del avión
    avion.mostrar_avion()

File: repos_estudiantes_TARQUI_segunda_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/COMPOSICION/EJERCICIO-3/Avion.java
import java.util.ArrayList;
import java.util.List;

public class Avion {
    private String modelo;
    private String fabricante;
    private List<Parte> partes;
    
    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    public String getModelo() {
        return modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public List<Parte> getPartes() {
        return partes;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public void setPartes(List<Parte> partes) {
        this.partes = partes;
    }
    
    public void agregarParte(Parte parte) {
        partes.add(parte);
    }
    
    public void mostrarAvion() {
        System.out.println("Modelo: " + modelo);
        System.out.println("Fabricante: " + fabricante);
        System.out.println("Partes del avion:");
        for (Parte p : partes) {
            p.mostrarInfo();
        }
    }
}
